src/ristretto/bulletproofs_plus.rs

```diff
@@ -0,0 +1,629 @@
+        let opening = CommitmentOpening::new(value, vec![key.0]);
+        let witness =
+            RangeWitness::init(vec![opening]).map_err(|e| RangeProofError::ProofConstructionError(e.to_string()))?;
+        let statement = RangeStatement::init(self.generators.clone(), vec![commitment], vec![None], None)
+            .map_err(|e| RangeProofError::ProofConstructionError(e.to_string()))?;
+
+        let proof = RistrettoRangeProof::prove(self.transcript_label, &statement, &witness)
+            .map_err(|e| RangeProofError::ProofConstructionError(e.to_string()))?;
+
+        Ok(proof.to_bytes())
+    }
+
+    fn verify(&self, proof: &Self::Proof, commitment: &HomomorphicCommitment<Self::PK>) -> bool {
+        return match RistrettoRangeProof::from_bytes(proof)
+            .map_err(|e| RangeProofError::InvalidRangeProof(e.to_string()))
+        {
+            Ok(rp) => {
+                let statement = RangeStatement {
+                    generators: self.generators.clone(),
+                    commitments: vec![commitment.0.clone().into()],
+                    commitments_compressed: vec![*commitment.0.compressed()],
+                    minimum_value_promises: vec![None],
+                    seed_nonce: None,
+                };
+                match RistrettoRangeProof::verify_do_not_recover_masks(self.transcript_label, &[statement], &[
+                    rp.clone()
+                ]) {
+                    Ok(_) => true,
+                    Err(e) => {
+                        if self.generators.extension_degree() != rp.extension_degree() {
+                            error!(
+                                target: LOG_TARGET,
+                                "Generators' extension degree ({:?}) and proof's extension degree ({:?}) do not \
+                                 match; consider using a BulletproofsPlusService with a matching extension degree",
+                                self.generators.extension_degree(),
+                                rp.extension_degree()
+                            );
+                        }
+                        error!(target: LOG_TARGET, "Internal range proof error ({})", e.to_string());
+                        false
+                    },
+                }
+            },
+            Err(e) => {
+                error!(
+                    target: LOG_TARGET,
+                    "Range proof could not be deserialized ({})",
+                    e.to_string()
+                );
+                false
+            },
+        };
+    }
+
+    fn range(&self) -> usize {
+        self.generators.bit_length()
+    }
+}
+
+impl ExtendedRangeProofService for BulletproofsPlusService {
+    type K = RistrettoSecretKey;
+    type PK = RistrettoPublicKey;
+    type Proof = Vec<u8>;
+
+    fn construct_extended_proof(
+        &self,
+        masks: Vec<RistrettoExtendedMask>,
+        values: Vec<u64>,
+        min_value_promises: Vec<Option<u64>>,
```

Comment:
```suggestion
        min_value_promises: Vec<u64>,
```

Reply:
As per previous comment: _The minimum value promise is optional, so although the default `None` will result in a lower bound proof of `0`, the intention of using the option is much clearer._